fix: "compare"-url generation now has a dash/hyphen in it

joundso opened the pull request:
Hi @kapsner,
the compare-URLs seem to have slightly changed:
https://gitlab.miracum.org/miracum/misc/diztools/-/commit/e950968000fcc98d8e809eb558ab8075466f0f93#8e91c39ea457744ee86e8378b154b7f3bb28e325_86_89

I stumbled across this when I wanted to upload a new CRAN release with the automatically created news.md and CRAN rejected due to the "moved url".

joundso committed:
fix: "compare"-url generation now has a slash in it

kapsner commented:
Thanks for this PR. Do you know if this adaption is still working for GitHub comparison-URLs, or does this need to be handled differently for Gitlab and GitHub?

kapsner commented:
Well, I guess I can answer this myself:

This is working:
[https://github.com/kapsner/autonewsmd/compare/v0.0.4...v0.0.5 ](https://github.com/kapsner/autonewsmd/compare/v0.0.4...v0.0.5)

While this is not working:
https://github.com/kapsner/autonewsmd/-/compare/v0.0.4...v0.0.5

I first need to figure out how to handle this properly.

kapsner committed:
fix: addresses issue with gitlab comparison found by @joundso

checks for presence of the keyword gitlab at the beginning of the repo-url;
if detected, replaces /compare/ with /-/compare/.

kapsner committed:
fix: moving from rmarkdown to quarto for rendering news.md

due to issues with pandoc on r-release-macos-x86_64 cran
' Error: pandoc version 1.12.3 or higher is required and was not found (see the help page ?rmarkdown::pandoc_available).'

kapsner commented on R/generate_autonewsmd.R:
```diff
@@ -186,8 +186,15 @@ generate_autonewsmd <- function(self, private) {
-        "`](", file.path(repo_url, "compare", set_changes), ")"
+        "`](", file.path(repo_url, "compare", set_changes, fsep = "/"), ")"
       )
+      if (grepl(pattern = "^(http(s)?://)?.*gitlab.*\\/", x = repo_url)) {
```
@joundso could you please check if this change solves your issue with the gitlab-comparison URL?

joundso replied:
Seems to work - thanks a lot!

kapsner replied:
Thx for the confirmation!

kapsner committed:
fix: fixed issue with presence of multiple associated remote repos

added interactive selection of the desired repository url

kapsner committed:
chore: added message to indicate location of copied output file

kapsner committed:
ci: added quarto installation to gha

kapsner committed:
ci: re-introduced pandoc installation for macos runners

kapsner committed:
chore: updated news.md
